### PR TITLE
feature: add selfci image builder job for bolt

### PR DESCRIPTION
**Pull request by blaisedias**
system test requires images suitable for self ci
i.e. testing of the tests themselves.
By definition, this is the last build that successfully,
ran the selfci test set.

### CODE REVIEW

File: pipelines/bolt/selfci-image-builder/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+pipeline {
+  agent any
+  options {
+    timeout(time: 180, unit: 'MINUTES')
+    buildDiscarder(logRotator(numToKeepStr: '365', artifactNumToKeepStr: '365'))
+  }
+  parameters {
+    string(defaultValue: '', name: 'image_tag', trim: true, description: 'image tag, auto-generated if build_images is set and this field is empty')
+    booleanParam(defaultValue: false, name: 'build_images', description: 'build mayastor images, one of dataplaneBranch or dataplaneCommitOrTag and one of controlplaneBranch or controlplaneCommitOrTag must be set.')
```
Comment (rohan2794) [inline pipelines/bolt/selfci-image-builder/Jenkinsfile:33]: is default value false correct here if this Jenkinsfile intended to build self-ci image for bolt ?

**Review by chriswldenyer — APPROVED** [review]
LGTM

File: pipelines/bolt/selfci-image-builder/Jenkinsfile (hunk 1 above)
Comment (blaisedias) [reply pipelines/bolt/selfci-image-builder/Jenkinsfile:33]: Good question.
I think yes, we should build images sparingly, ideally just use the one built overnight.
Then we have a more consistent view of status.
The plan is to invoke the selfci image builder from the daily and weekly jobs.
I should add a validation stage, so as to avoid invoking generic-system-test with an empty `image_tag` and `build_images` set to `false`.

Comment (blaisedias) [reply pipelines/bolt/selfci-image-builder/Jenkinsfile:33]: @rohan2794  @chriswldenyer  I see an issue with the name, maybe this should be `selfci-image` rather than `selfci-image-builder`?

**Review by rohan2794 — APPROVED** [review]
LGTM
